src/Symfony/Bridge/Twig/Extension/HttpFoundationExtension.php

```diff
@@ -24,13 +25,27 @@
  */
 class HttpFoundationExtension extends AbstractExtension
 {
-    private $requestStack;
-    private $requestContext;
+    private $urlHelper;
 
-    public function __construct(RequestStack $requestStack, RequestContext $requestContext = null)
+    public function __construct($urlHelper, RequestContext $requestContext = null)
     {
-        $this->requestStack = $requestStack;
-        $this->requestContext = $requestContext;
+        if ($urlHelper instanceof UrlHelper) {
+            $this->urlHelper = $urlHelper;
+
+            return;
+        }
+
+        if (!$urlHelper instanceof RequestStack) {
+            throw new \InvalidArgumentException(sprintf('The first argument must be an instance of "%s" or an instance of "%s".', UrlHelper::class, RequestStack::class));
```

Comment:
You could throw a `TypeError`. This would be consistent withe previous behavior.